x-pack/plugins/cloud_security_posture/public/common/api/use_latest_findings_data_view.ts

```diff
@@ -8,22 +8,75 @@
 import { useQuery } from '@tanstack/react-query';
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import type { DataView } from '@kbn/data-plugin/common';
+import { i18n } from '@kbn/i18n';
+import { LATEST_FINDINGS_INDEX_PATTERN } from '../../../common/constants';
 import { CspClientPluginStartDeps } from '../../types';
 
+const cloudSecurityFieldLabels: Record<string, string> = {
+  'result.evaluation': i18n.translate(
+    'xpack.csp.findings.findingsTable.findingsTableColumn.resultColumnLabel',
+    {
+      defaultMessage: 'Result',
+    }
+  ),
+  'resource.id': i18n.translate(
+    'xpack.csp.findings.findingsTable.findingsTableColumn.resourceIdColumnLabel',
+    { defaultMessage: 'Resource ID' }
+  ),
+  'resource.name': i18n.translate(
+    'xpack.csp.findings.findingsTable.findingsTableColumn.resourceNameColumnLabel',
+    { defaultMessage: 'Resource Name' }
+  ),
+  'resource.sub_type': i18n.translate(
+    'xpack.csp.findings.findingsTable.findingsTableColumn.resourceTypeColumnLabel',
+    { defaultMessage: 'Resource Type' }
+  ),
+  'rule.benchmark.rule_number': i18n.translate(
+    'xpack.csp.findings.findingsTable.findingsTableColumn.ruleNumberColumnLabel',
+    {
+      defaultMessage: 'Rule Number',
+    }
+  ),
+  'rule.name': i18n.translate(
+    'xpack.csp.findings.findingsTable.findingsTableColumn.ruleNameColumnLabel',
+    { defaultMessage: 'Rule Name' }
+  ),
+  'rule.section': i18n.translate(
+    'xpack.csp.findings.findingsTable.findingsTableColumn.ruleSectionColumnLabel',
+    { defaultMessage: 'CIS Section' }
+  ),
+  '@timestamp': i18n.translate(
+    'xpack.csp.findings.findingsTable.findingsTableColumn.lastCheckedColumnLabel',
+    { defaultMessage: 'Last Checked' }
+  ),
+} as const;
+
 /**
  *  TODO: use perfected kibana data views
  */
 export const useLatestFindingsDataView = (dataView: string) => {
   const {
-    data: { dataViews },
+    data: { dataViews, indexPatterns },
   } = useKibana<CspClientPluginStartDeps>().services;
 
   const findDataView = async (): Promise<DataView> => {
-    const dataViewObj = (await dataViews.find(dataView))?.[0];
+    const [dataViewObj] = await dataViews.find(dataView);
     if (!dataViewObj) {
       throw new Error(`Data view not found [Name: {${dataView}}]`);
     }
 
+    if (dataView === LATEST_FINDINGS_INDEX_PATTERN) {
+      Object.entries(cloudSecurityFieldLabels).forEach(([field, label]) => {
+        if (
+          !dataViewObj?.getFieldAttrs()?.[field]?.customLabel ||
+          dataViewObj.getFieldAttrs()[field].customLabel === field
+        ) {
```

Comment:
```suggestion
        if (
          !dataViewObj.getFieldAttrs()[field]?.customLabel ||
          dataViewObj.getFieldAttrs()[field]?.customLabel === field
        ) {
```

We already checked the dataViewObj exists on a previous `if`, so there's no need to double check that. also according to its type `getFieldAttrs` should exist, but i think adding the check after the field name is viable since if there no such field we will crash.

Reply:
thanks for the suggestion